xarray/coding/times.py

```diff
@@ -777,6 +777,11 @@ def encode(self, variable: Variable, name: T_Name = None) -> Variable:
             safe_setitem(attrs, "units", units, name=name)
             safe_setitem(attrs, "calendar", calendar, name=name)
 
+            # drop dtype in encoding if encoded as float64
+            # to prevent unnecessary casts, see GH #1064
+            if np.issubdtype(data.dtype, np.float64):
```

Comment:
I think we may only want to do this in the case that the user-provided encoding is of integer dtype (otherwise we will prevent user-prescribed encoding dtypes of lower precision floating point values like `np.float32`, etc.).  It could be good to add a test for this.
```suggestion
            encoding_dtype = encoding.get("dtype", data.dtype)
            if np.issubdtype(data.dtype, np.float64) and encoding_dtype.kind in "iu":
```